File: Promact.CustomerSuccess.Platform/Entities/VersionHistory.cs
using System;
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;
using Volo.Abp.Domain.Entities.Auditing;

namespace Promact.CustomerSuccess.Platform.Entities
{
    public class VersionHistory : AuditedAggregateRoot<Guid>
    {
        [ForeignKey("Project")]
        public Guid ProjectId { get; set; }
        public virtual Project? Project { get; set; }
        public required string Version { get; set; }
        public required string Type { get; set; }
        public required string Change { get; set; }
        public required string ChangeReason { get; set; }
        public required string CreatedBy { get; set; }
        public DateTime RevisionDate { get; set; }
        public DateTime? ApprovalDate { get; set; }
        public required string ApprovedBy { get; set; }

    }
}

File: Customer_Success_Platform_Client/src/app/Models/VersionHistory.ts
export interface VersionHistory {
    id: string;
    projectId:string;
    version: string;
    type: string;
    change: string;
    changeReason: string;
    createdBy: string;
    revisionDate: Date;
    approvalDate?: Date;
    approvedBy: string;
  }